universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -159,40 +164,93 @@ public Program createOrUpdateDraft(Program existingProgram) {
     return database
         .find(Program.class)
         .where()
-        .eq(
-            "name",
-            database.find(Program.class).setId(programId).select("name").findSingleAttribute())
+        .in("name", database.find(Program.class).select("name").where().eq("id", programId).query())
+        .query()
         .findList()
         .stream()
         .collect(ImmutableList.toImmutableList());
   }
 
-  public ImmutableList<Long> getAllProgramVersionIds(long programId) {
+  public PaginationResult<Application> getApplicationsForAllProgramVersions(
+      long programId, PaginationSpec paginationSpec, Optional<String> searchNameFragment) {
+    ExpressionList<Application> query =
+        database
+            .find(Application.class)
+            .fetch("program")
+            .orderBy("id desc")
```

Comment:
method doc that they're returned in desc order; or add it to the method name, etc.  It seems to matter but it's not part of the "contract"

Reply:
it ensures they're submitted in the order they were created, so submitted. could use submit time instead but for testing that gets dicey when we create several too quickly.

Reply:
sgtm, just needed to be doc'd if it was imp.